pytorch_lightning/strategies/launchers/subprocess_script.py

```diff
@@ -0,0 +1,118 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import subprocess
+import sys
+from subprocess import Popen
+from time import sleep
+from typing import Any, Callable, List, Optional
+
+import __main__
+import numpy as np
+
+from pytorch_lightning.plugins.environments.cluster_environment import ClusterEnvironment
+from pytorch_lightning.strategies.launchers.base import _Launcher
+from pytorch_lightning.utilities import _HYDRA_AVAILABLE
+
+if _HYDRA_AVAILABLE:
+    from hydra.core.hydra_config import HydraConfig
+    from hydra.utils import get_original_cwd, to_absolute_path
+
+
+class _SubprocessScriptLauncher(_Launcher):
+    r"""
+    Creates and launches subprocess scripts on each device.
+    """
+
+    def __init__(self, cluster_environment: ClusterEnvironment, num_processes: int, num_nodes: int) -> None:
+        super().__init__()
+        self.cluster_environment = cluster_environment
+        self.num_processes = num_processes
+        self.num_nodes = num_nodes
+        self.interactive_ddp_procs: List[Popen] = []
```

Comment:
nit: no longer need to keep ddp in the name

Reply:
Good catch! Btw do you know why we need to store process in it? I don't see it being used anywhere.

Reply:
I'm not sure of its history (n00b git question: how do you track the git blame across renames?)

was the intention for the strategy to terminate the subprocess at the end?